test/hpr_config_worker_SUITE.erl

```diff
@@ -303,6 +309,32 @@ delete_route_test(Config) ->
     ?assertEqual(0, ets:info(hpr_config_routes, size)),
 
     ok.
+
+backoff_test(Config) ->
+    %% Let it startup
+    timer:sleep(100),
+
+    ServerPid = proplists:get_value(server_pid, Config),
+
+    meck:new(backoff, [passthrough]),
+
+    gen_server:stop(ServerPid),
+
+    ?assertEqual(1, meck:num_calls(backoff, fail, 1)),
+
+    %% We sleep 3s the BACKOFF_MIN is 1s so we should get another 2 tries
+    timer:sleep(3000),
```

Comment:
If can assume that `backoff` works properly, you could make this test run faster by mocking `backoff` and have it return canned values. If you do this, you won't need to the `TEST` definitions for the timeouts. It might be easier to also mock `grpc_client:connect` to simulate the failure instead of stopping the server.

Reply:
I am not a massive fan of mocking all thew time I would rather wait an extra few seconds and get the "real" behavior.